src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -365,8 +365,43 @@ public static bool TryText(FormattingInfo formatInfo, IRContext irContext, Formu
                 return false;
             }
 
-            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
-
+            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
+
+            bool hasFormatCulture = false;
+            CultureInfo formatCulture = culture;
+            CultureInfo enUsCulture = CultureInfo.GetCultureInfo("en-US");
+            string revFormatStr = formatString;
+
+            if (!string.IsNullOrEmpty(formatString))
+            {
+                int startIdx = formatString.IndexOf("[$-", StringComparison.OrdinalIgnoreCase);
+                int endIdx = formatString.IndexOf("]", StringComparison.OrdinalIgnoreCase);
+
+                if (startIdx > -1 && endIdx > 2)
+                {
+                    startIdx += 3;
+                    hasFormatCulture = true;
+                    string formatCultureName = formatString.Substring(startIdx, endIdx - startIdx);
+
+                    if (!TryGetCulture(formatCultureName, out formatCulture))
+                    {
+                        return false;
+                    }
+
+                    var numberCultureFormat = formatCulture.NumberFormat;
+                    formatString = formatString.Substring(endIdx + 1);                    
+                    revFormatStr = formatString.Replace(numberCultureFormat.NumberGroupSeparator, "\uFEFF");
+
+                    if (string.IsNullOrWhiteSpace(numberCultureFormat.NumberGroupSeparator))
+                    {
+                        revFormatStr = revFormatStr.Replace(" ", "\uFEFF").Replace("\u202F", "\uFEFF");

```

Comment:
>"\uFEFF" [](http://example.com/codeflow?start=65&length=8)

nit: can we assign this to a const string field, and document what it means. That will make it easier to read code (and verify all of these literals are indeed the same unicode char)

Reply:
updated